### ISSUE TITLE
Giseki - Rock Crab

### ISSUE THREAD

**Issue by CptA4** [MineInAbyss/Issue-Tracker#135]
### Mob name

Giseki

### Spawn Locations

Layer 5

### Mob-type

Hostile

### Mob Info

Serious in difficulty is appropriate, it spawns on either grass blocks or stone. This mob spawns as a block which is as hard as obsidian _(If possible it should take longer to mine that obsidian)_ but visually resembles stone with a very, very, slight green hue at the bottom to indicate moss and distinction from normal stone.

There are two way to make a Giseki revert from it's block state, if a player is within 2 blocks of the Giseki's block state, the actual mob will spawn in while the block will be destroyed, If the mob doesn't detect the player after a set time, it will revert to it's block state. The second way to revert it from a block state is to break it's block state, after which it will spawn like normal with half it's normal health. The Giseki's Hardness comes from a hard crystal which makes up it's insides while it's exterior is a thin layer of stone, Upon being killed the Giseki will drop crystals which resemble Opal. (Drop rates discussed later)

As for special attacks, once it attacks the player for the first time, it won't "let go" of the player, this can probably be simulated by having the mob somehow be closely attached to the player, once it's grabbed the player it will deal 10 base damage every 3 seconds, if the Giseki is hurt, it will instantly deal damage to the player in response, the main way to get them to stop grabbing the player is to kill it. Once a player is grappled they are unable to jump and can't walk or run at full speed, much like slowness.
It would also be cool if it detects two players close-by it will automatically go back into it's block state even if it's currently grabbed onto a player. 

_[Past this point is very interesting ideas but I'm not sure entirely sure if they're possible]_

Gisekis will change stages every 8 minutes they're alive, with 4 stages and 8 minutes per stage will add up to a total of 32 minutes of wait time to reach the final stage, stage 4, the first stage is Full Stone, giving the player 0-2 crystals, Mostly Stone, 2-3 crystals, Mostly Crystal, 3-6 crystals, Full Crystal, 6-18 crystals. Along with the stages of the Giseki, it's texture will change overtime.

- Stage 1: 0-2
- Stage 2: 2-3
- Stage 3: 3-6
- Stage 4: 6-18

Also it would be funny for the mob to be ridable, and while riding it the player will not be attacked by the Giseki.

### Mob Model/Texture

![giseki stages](https://user-images.githubusercontent.com/48418904/168154562-a9e34578-87e6-4ed6-aed7-309688438084.png)
![lil dud](https://user-images.githubusercontent.com/48418904/168154568-3de8fc2a-73cc-487b-911c-85f2f1d58799.png)

### Agreements

- [X] I have searched for and ensured there isn't already an open suggestion similar to this.
- [X] I have ensured the mob I'm suggesting isn't already on the server.

**Comment by VaIienta** [comment]
Sounds cool to have a mob which changes between a block and a mob like that, not entirely sure what the code implementation would be for that but it would definitely require mob behaviours we don't have at the moment (I'm guessing).

One other issue could be spawning, don't want to it to spawn in player builds for example - but if this was aimed towards adventure then that wouldn't be an issue.

Not sure how we would also do the whole grappling behaviour, but it does sound more interesting than just attacking - I wouldn't worry too much about how much dmg it deals at the moment it's better to scale that with testing in game.

Visual design wise looks really good, I know Luky had been making a model I do not know if he finished it.

And talking about variance I suppose this could be an opportunity for it, like maybe having one that is stone and sand or stone and snow?

Not too sure about the stage thing, are you suggesting over time it increases crystals?

Think there should be some proper visual indicator it's  a mob when it's hiding as a "block", for example say it's in the floor - it would be impossible to tell. Maybe something on its head as a visual warning?

**Comment by getvood** [comment]
I think the mob can be adventure and survival, since it's block form would automatically be destroyed once the actual mob spawns in it's place that shouldn't be a problem.

Not too sure about variants on the coding side but that seems fun.

And yeah with the stages that's what I mean, the amount of crystals that drop will increase exponentially, alongside with a visual indicator of the crystals growing inside of the mob but the crystals eventually have no more room to grow so it slowly grows through the outside of the shell.

As for a visual indicator, it's block form would be stone with a very slight hint of moss growth around the corners, something very subtle where it can't be instantly recognized but if someone took 2 seconds to look, they would be informed of the difference

**Comment by VaIienta** [comment]
Yeah I suppose on the spawning side of it I was thinking it would replace a block which would make it seem more natural. If it were to spawn then yeah you would just have to check it was above stone or something? (Otherwise it could just fall down) though the only issue you would get is like say a player made a stone statue you would randomly get them spawning on that... But that's just some scenarios

Kind of feel like this needs a Dev opinion on the spawning side of things mainly to check actually how this could be implemented.. whether it would be blocks or all entities just kinda making sure it's do-able in a nice way.

As for the crystals increasing exponentially that sounds a bit grindy and abusable and I'm not sure we can do textures that actively change on mobs. Maybe like some trade-off would be better? Maybe you give it some coal or stone (idk leave it on the ground while it's not hostile or something) and it increases the amount of crystals but it will only take a set amount (as to not make it exponential). Maybe something down that road where it gets the player more involved and stops it being exponential and less grindy?

**Comment by getvood** [comment]
For the spawning I was thinking instead of it replacing a stone block, I guess to explain how it may work on the code-side of things, it would spawn, then a custom stone block is placed ontop of it's position and then if the custom block is broken or if someone goes into it's "activation range", then the custom block will be destroyed. It's just what I'm thinking in my head but I'm sure the code will have a different method. 

For the stages, maybe instead of an increase in crystals, every Giseki can spawn at a random stage, so 4 different textured models that don't change.
